### PR TITLE
Feature/init comm

### PR DESCRIPTION
**Pull request by bjpalmer**
Merge comm-init branch into develop. This adds capability to initialize GA from a communicator other than MPI_COMM_WORLD

### CODE REVIEW

**Review by ajaypanyala — APPROVED** [review]

**Comment by bjpalmer** [comment]
@ajaypanyala, these failures appear to originate from a build that is linking to an external ARMCI library. Is this correct? If so, the fix would be to add an ARMCI_Init_mpi_comm function to the external library. Do you know where we are getting that library from?

**Comment by jeffhammond** [comment]
It fails with ARMCI-MPI, which is the only other implementation of ARMCI, as far as I know. 

I'll add this function to ARMCI-MPI today.

File: global/testing/CMakeLists.txt [hunk 1]
```diff
@@ -9,8 +9,7 @@
 # agency of the United States Government.  Neither the United States
 # Government nor the United States Department of Energy, nor Battelle,
 # nor any of their employees, MAKES ANY WARRANTY, EXPRESS OR IMPLIED, OR
-# ASSUMES ANY LEGAL LIABILITY OR RESPONSIBILITY FOR THE ACCURACY,
-# COMPLETENESS, OR USEFULNESS OF ANY INFORMATION, APPARATUS, PRODUCT,
+# ASSUMES ANY LEGAL LIABILITY OR RESPONSIBILITY FOR THE ACCURACY, # COMPLETENESS, OR USEFULNESS OF ANY INFORMATION, APPARATUS, PRODUCT,
```
Comment (jeffhammond) [inline global/testing/CMakeLists.txt:12]: you should undo this.  `# COMPLETENESS` belongs on a new line.

**Comment by jeffhammond** [comment]
ARMCI-MPI uses `ARMCI_Init_thread_comm(int armci_requested, MPI_Comm comm)` as the real initialization function, so it was trivial to add `ARMCI_Init_mpi_comm` as another entry point.  It's [merged](https://github.com/pmodels/armci-mpi/pull/43) so if you restart the tests, they should pass.

**Pull request merged**